### ISSUE TITLE
new installation of 0.4.2 asks for upgrade

### ISSUE THREAD

**Issue by realsobek** [bibliocoll/JournalTouch#86]
Using a new installation of JournalTouch 0.4.2 on CentOS 7 x86_64 the first visit of:
`http://journaltouch/`
greets with:
`JournalTouch has to be upgraded. Please go to Admin Upgrader`
'Admin Upgrade' links to:
`http://journaltouch/admin/upgrade.php`

I visited said page. Output was:
```
Updating JournalTouch
Updating to version 0.4.2
Folder check: success
Successfully upgraded to: 0.4.2
```

Can I trigger the upgrade in automatic unattended installation in order to avoid manual action?

**Comment by tzeumer** [comment]
> Can I trigger the upgrade in automatic unattended installation in order to avoid manual action?

Currently the easiest way to prevent this, is to add this (empty) file: data/upgraded/ver_0.4.2. That's the short answer. The long one...

The upgrader is a pretty dumb thing. It just checks if for the version set in sys/bootstrap.php a file exists in data/upgraded. The idea is that on a "real" upgrade the upgrade actions for each missing version since the last user side version are run. 

This was especially "important" for the switch from 0.3 to 0.4 since there were many changes in the folder structure and I didn't want anyone to have to think about manually moving there user files or deleting old folders.(I especially wanted to convince the maintainers to give up the old folder structure ;)).

The problem (I missed) is that for a new installation the "current version" file should not be missing in data/upgraded. Yet it has to be missing for an upgrade. The only way would be to create 2 releases: update and new. 

I guess there are much more sophisticated ways for upgrading stuff. I think it even wouldn't be so hard to implement an upgrade via backend. But since we now (nearly) separated all user data finally the update procedure could just be "delete everything but the data folder and copy the new version over it" - that's prettty much like wordpress does/did it.

**Comment by krugar** [comment]
> But since we now (nearly) separated all user data finally the update procedure could just be "delete everything but the data folder and copy the new version over it" - that's prettty much like wordpress does/did it.

speaking of which... :) any wishes as to where `/data/config/config-default.php` should end up? i'd move it to `/sys` if you don't have a better idea?

**Comment by krugar** [comment]
if this happens to you with 0.4.4, you may want to add an empty file at `data/upgraded/ver_0.4.4`

**Comment by krugar** [comment]
@buczek provided a fix for this, merged into `develop` at 5506d92. will test and amend as needed, solution goes into 0.4.5
